### PR TITLE
dont hide stack trace for root exception

### LINKED ISSUE #10464: Dont hide exception stacktrace in packtask
## Details about Problem

When a top level exception occurs in packtask, the stack trace is hiddin. this makes it difficult to work out what the underlying issue is.

image

## Product version: 

see environment below

## Repro steps and/or sample project

difficult to create a repro when cant work out what the issue it

## environment:

```
Microsoft Visual Studio Professional 2019 Preview
Version 16.9.0 Preview 2.0
VisualStudio.16.Preview/16.9.0-pre.2.0+30803.129
Microsoft .NET Framework
Version 4.8.03761

Installed Version: Professional

ASP.NET and Web Tools 2019   16.9.311.8738
ASP.NET and Web Tools 2019

ASP.NET Core Razor Language Services   16.1.0.2057006+e0968ea4b5f5c523cc7efea1d91fec21e0adc648
Provides languages services for ASP.NET Core Razor.

ASP.NET Web Frameworks and Tools 2019   16.9.311.8738
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.9.311.8738
Azure App Service Tools v3.0.0

C# Tools   3.9.0-2.20602.2+eb0a7ff1c241695e17b07b096593804fdd5cfcfd
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

JetBrains ReSharper 2020.3.2   Build 203.0.20201229.111204
JetBrains ReSharper package for Microsoft Visual Studio. For more information about ReSharper, visit http://www.jetbrains.com/resharper. Copyright © 2021 JetBrains, Inc.

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

NuGet Package Manager   5.9.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.62011.24130
Microsoft SQL Server Data Tools

TypeScript Tools   16.0.21120.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.9.0-2.20602.2+eb0a7ff1c241695e17b07b096593804fdd5cfcfd
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   16.9.0-beta.20567.7+c02b2ef997f1dd07993e1f85b017c858bdf82530
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers
```

### PR DESCRIPTION
**Pull request by SimonCropp**
## Bug

Fixes: https://github.com/NuGet/Home/issues/10464
Regression: No  

## Fix

include stack trace

## Testing/Validation

Tests Added: No  
Reason for not adding tests:  should be covered by existing tests
Validation: no idea how to do this

### CODE REVIEW

**Comment by donnie-msft** [comment]
Hi @SimonCropp, for the validation, can you try `Pack` in the same way you did before, and attach a screenshot of how the new error looks with the stack?
Docs for how to build a VSIX are here https://github.com/NuGet/NuGet.Client/blob/dev/docs/debugging.md#testing-the-build-of-nuget-in-visual-studio, and you can test it in the VS Experimental Instance.

File: src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs [hunk 1]
```diff
@@ -129,7 +129,7 @@ public override bool Execute()
             }
             catch (Exception ex)
             {
-                ExceptionUtilities.LogException(ex, Logger);
+                ExceptionUtilities.LogException(ex, Logger, true);
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs:132]: What happens when a PackagingException is thrown?

In particular take this project: 

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <IncludeBuildOutput>false</IncludeBuildOutput>
    <TargetFramework>net5.0</TargetFramework>
  </PropertyGroup>

</Project>
```

Today if you run it, it logs:

```
C:\Program Files\dotnet\sdk\5.0.200-preview.20601.7\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(221,5): error NU5017: Cannot create a package that has no dependencies nor content. [C:\Users\Nikolche\Source\Repos\PR3853\PR3853.csproj]
```

**Review by nkolev92 — COMMENTED** [review]
Commented on the issue too, but I think the stack trace should already be logged at detailed verbosity.

**Comment by SimonCropp** [comment]
@donnie-msft 

> Hi @SimonCropp, for the validation, can you try Pack in the same way you did before, and attach a screenshot of how the new error looks with the stack? Docs for how to build a VSIX are here https://github.com/NuGet/NuGet.Client/blob/dev/docs/debugging.md#testing-the-build-of-nuget-in-visual-studio, and you can test it in the VS Experimental Instance.

Sorry but that is more work than i care to put into this annoyance

**Pull request closed without merging**